### PR TITLE
feat: disable ETH ERC20 features

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Disables ERC20 and WETH9 features in `OVM_ETH`, moves the contract to `0xdeaddead....0000`. 

**Metadata**
- Fixes ENG-1365

### CODE REVIEW

**Comment by smartcontracts** [comment]
TODO: add tests for `OVM_ETH`, specifically checking that ERC20 features are not accessible.

File: packages/contracts/contracts/L2/predeploys/OVM_ETH.sol [hunk 1]
```diff
@@ -31,52 +30,26 @@ contract OVM_ETH is L2StandardERC20, IWETH9 {
     {}
 
 
-    /******************************
-     * Custom WETH9 Functionality *
-     ******************************/
-    fallback() external payable {
-        deposit();
+    // Temporarily disable ETH ERC20 features.
```
Comment (tynes) [inline packages/contracts/contracts/L2/predeploys/OVM_ETH.sol:33]: This isn't temporary if this gets merged?

File: packages/contracts/contracts/L2/predeploys/OVM_ETH.sol [hunk 2]
```diff
@@ -31,52 +30,26 @@ contract OVM_ETH is L2StandardERC20, IWETH9 {
...
+    // Discussion here: https://github.com/ethereum-optimism/optimism/discussions/1444
+
+    function transfer(address recipient, uint256 amount) public virtual override returns (bool) {
+        revert("OVM_ETH: ERC20 features are disabled pending further community discussion.");
```
Comment (tynes) [inline packages/contracts/contracts/L2/predeploys/OVM_ETH.sol:37]: This should be updated to say `OVM_ETH: transfer is disabled`

Comment (tynes) [reply packages/contracts/contracts/L2/predeploys/OVM_ETH.sol:37]: This applied for each additional revert below

**Comment by tynes** [comment]
Is the idea of this PR to make it so that ETH balance shows up in the account itself instead of being in a contract? If so, I believe this needs to be updated as well. https://github.com/ethereum-optimism/optimism/blob/8dda91206e7b3b2a8f9839e278c90873f973594f/l2geth/core/state/statedb.go#L369-L379

**Comment by smartcontracts** [comment]
Balance will still be in the contract until further notice. Currently we need this in order to do ETH withdrawals but that will likely change in the update after this next one.

File: packages/contracts/contracts/L2/predeploys/OVM_ETH.sol (hunk 2 above)
Comment (K-Ho) [reply packages/contracts/contracts/L2/predeploys/OVM_ETH.sol:37]: Can we just make these functions private or just delete them altogether? I don't even think we should tell developers about ETH existing at this `deaddeaddead...` address (maybe this info exists in our specs, but not in our developer docs)

File: packages/contracts/test/contracts/L2/predeploys/OVM_ETH.spec.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  describe('transfer', () => {
+    it('should revert', async () => {
+      await expect(
+        OVM_ETH.transfer(await signer2.getAddress(), 100)
+      ).to.be.revertedWith(
+        'OVM_ETH: ERC20 features are disabled pending further community discussion.'
+      )
+    })
+  })
+
```
Comment (K-Ho) [inline packages/contracts/test/contracts/L2/predeploys/OVM_ETH.spec.ts:33]: Same comment as above - do we even need to have these functions exist? We can just delete them + these tests for now

**Review by K-Ho — COMMENTED** [review]
Looks great! Just would suggest removing all of the ERC20 functions altogether instead of reverting for them

**Comment by smartcontracts** [comment]
> Looks great! Just would suggest removing all of the ERC20 functions altogether instead of reverting for them

Unfortunately I can't do this without making a huge diff because OVM_ETH inherits L2StandardERC20 which inherits ERC20 etc etc. Disabling the functions was the easiest option.

**Comment by K-Ho** [comment]
> > Looks great! Just would suggest removing all of the ERC20 functions altogether instead of reverting for them
> 
> Unfortunately I can't do this without making a huge diff because OVM_ETH inherits L2StandardERC20 which inherits ERC20 etc etc. Disabling the functions was the easiest option.

Didn't think of that - will go ahead and approve 👍

**Review by K-Ho — APPROVED** [review]

**Pull request merged**